### ISSUE TITLE
Win 10 / Python 2.7 bad cache issue - Fixed

### ISSUE THREAD

**Issue by RK1975** [r0x0r/pywebview#284]
I so wanted this to be a lib I could just install and use...  It would be so helpful.

But it's an absolute nightmare.  First of all, trying to get PyQt4 running in Python 2.7 when the author has introduced fun steps like having to compile it yourself in a bid to encourage migration to Python 3.x...

Is there no other way to get a web view other than to use PyQt?
The concept is awesome.  The dependencies drove me nuts and after several hours I gave up.

**Comment by shivaprsdv** [comment]
Sorry, can't really understand your issue. If you are on Linux it runs on both GTK
and QT. If you are using QT, it would be better to use QT5.

**Comment by RK1975** [comment]
I'm on Python 2
7 on Win 10.  I know a lot of Python devs wish the world was on Linux but
they're not exclusively.
As I said, you have to get QT on first which is painful as the QT author
doesn't support Python 2.7 with an easy to install lib...  So I can't get
your project working due to all the dependencies.
I've installed loads of libs over the years with dependencies but this is
just painful with the combo of 2.7 on Win.

**Comment by shivaprsdv** [comment]
Oh, you _don't_ need QT on Windows. What you need is `pythonnet`. I don't
if it is easy to install or not. Please see [installation](https://pywebview.flowrl.com/guide/installation.html#dependencies) page.

**Comment by RK1975** [comment]
Thank you.  Just tried it at home.  Worked!  -- however, at work it thinks
I need PyQT5:

[image: image.png]

**Comment by shivaprsdv** [comment]
What system do you use at work? Is it Linux or Windows?

**Comment by RK1975** [comment]
This is what happens when it tries to install on the work machine - it tries to get Qt5:

Collecting pywebview
Collecting pythonnet (from pywebview)
  Downloading https://files.pythonhosted.org/packages/4f/28/e5bed7fe232d46c62bc64027071bc9f525ef1fd30dcd6fa1031891079b1b/pythonnet-2.3.0-cp27-cp27m-win32.whl (58kB)
    100% |################################| 61kB 2.0MB/s
Collecting pyobjc (from pywebview)
  Using cached https://files.pythonhosted.org/packages/a3/db/c37b79ce4c58e90dfa3cb0403ba7dc8238fc2d8614db0377bcaf85efc989/pyobjc-5.1.1-py2-none-any.whl
Collecting PyQt5 (from pywebview)
  Could not find a version that satisfies the requirement PyQt5 (from pywebview) (from versions: )
No matching distribution found for PyQt5 (from pywebview)

**Comment by r0x0r** [comment]
This is a bug in a previous version, which has been fixed. It seems like
your pip has got the previous version cached and tries to install it
instead.
`pip install pywebview-U ` should do the trick.

**Comment by RK1975** [comment]
Fixed on work's PC now 👍 

`pip install pywebview --no-cache-dir`

stops it trying to use any broken cached stuff (on pip V>6.0)

guys - thanks for all your help.  I've been playing with this and it is stunning, it's going to make development so much easier for me.  I tried adding a button dynamically to a page and then calling the some python code.  Works perfectly - I am going to love using this.  Nice work, it makes powerful GUIs on Python a breeze,  Awesome work @r0x0r  / @shivaprsdv
